MS provider post-deployment check failed

The post-deployment of the provider cluster failed in Managed Services.
```
>       assert verify_worker_nodes_security_groups()
E       AssertionError

ocs_ci/ocs/resources/storage_cluster.py:1378: AssertionError
```
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/17236/testReport/

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -18,7 +18,7 @@ from ocs_ci.ocs.bucket_utils import craft_s3_command
 from ocs_ci.ocs.ocp import get_images, OCP, verify_images_upgraded
 from ocs_ci.helpers import helpers
 from ocs_ci.helpers.proxy import update_container_with_proxy_env
-from ocs_ci.ocs import constants, defaults, workload, ocp
+from ocs_ci.ocs import constants, defaults, node, workload, ocp
 from ocs_ci.framework import config
 from ocs_ci.ocs.exceptions import (
     CephToolBoxNotFoundException,
@@ -1497,6 +1497,7 @@ def get_pod_node(pod_obj):
     node_name = pod_obj.get().get("spec").get("nodeName")
     if not node_name:
         raise NotFoundError(f"Node name not found for the pod {pod_obj.name}")
+    return node.get_node_objs(node_names=node_name)[0]
 
 
 def delete_pods(pod_objs, wait=True):

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -28,6 +28,7 @@ from ocs_ci.ocs.exceptions import (
     TimeoutExpiredError,
     UnavailableResourceException,
     ResourceNotFoundError,
+    NotFoundError,
 )
 from ocs_ci.ocs.utils import setup_ceph_toolbox, get_pod_name_by_pattern
 from ocs_ci.ocs.resources.ocs import OCS
@@ -1489,10 +1490,13 @@ def get_pod_node(pod_obj):
     Returns:
         ocs_ci.ocs.ocp.OCP: The node object
 
+    Raises:
+        NotFoundError when the node name is not found
+
     """
     node_name = pod_obj.get().get("spec").get("nodeName")
-    assert node_name, f"Node name not found for the pod {pod_obj.name}"
-    return node.get_node_objs(node_names=node_name)[0]
+    if not node_name:
+        raise NotFoundError(f"Node name not found for the pod {pod_obj.name}")
 
 
 def delete_pods(pod_objs, wait=True):

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -1491,6 +1491,7 @@ def get_pod_node(pod_obj):
 
     """
     node_name = pod_obj.get().get("spec").get("nodeName")
+    assert node_name, f"Node name not found for the pod {pod_obj.name}"
     return node.get_node_objs(node_names=node_name)[0]
 
 

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -18,7 +18,7 @@ from ocs_ci.ocs.bucket_utils import craft_s3_command
 from ocs_ci.ocs.ocp import get_images, OCP, verify_images_upgraded
 from ocs_ci.helpers import helpers
 from ocs_ci.helpers.proxy import update_container_with_proxy_env
-from ocs_ci.ocs import constants, defaults, node, workload, ocp
+from ocs_ci.ocs import constants, defaults, workload, ocp
 from ocs_ci.framework import config
 from ocs_ci.ocs.exceptions import (
     CephToolBoxNotFoundException,

